tests/common.py

```diff
@@ -30,7 +30,7 @@
 #   in case a mistake in the code lead to a deadlock :(
 # So the solution is to make this configurable: a good middleground by default
 # and a long long time on the CI.
-_SIDE_EFFECTS_TIMEOUT = 3
+_SIDE_EFFECTS_TIMEOUT = 4
```

Comment:
3s on a regular pc should be huge (we increase this to 10s on CI just to be sure), in theory we should be able to keep this to 1s or even lower.
So increasing this seems really smelly... what the reason ?

Reply:
I tried to run the tests in parallels :sweat_smile:

Reply:
There is definitely something fishy here, you should investigate what is taking so long